### PR TITLE
feat: support chat completion in studio SDK

### PR DESCRIPTION
**Pull request by asafgardin**

### CODE REVIEW

File: examples/studio/chat.py [hunk 1]
```diff
@@ -8,11 +9,12 @@
     ChatMessage(text="I am having trouble signing up for your product with my Google account.", role=RoleType.USER),
 ]
 
+
 client = AI21Client()
```
Comment (miri-bar) [inline examples/studio/chat.py:13]: Might be worth adding a comment here saying that this is deprecated and reference to the new method

File: examples/studio/chat_complete.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+from ai21.models import ChatMessage, RoleType
+
+system = "You're a support engineer in a SaaS company"
+messages = [
+    ChatMessage(content="Hello, I need help with a signup process.", role=RoleType.USER),
+    ChatMessage(content="Hi Alice, I can help you with that. What seems to be the problem?", role=RoleType.ASSISTANT),
+    ChatMessage(content="I am having trouble signing up for your product with my Google account.", role=RoleType.USER),
+]
+
+AI21EnvConfig.api_host = "https://api-stage.ai21.com"
```
Comment (miri-bar) [inline examples/studio/chat_complete.py:11]: Delete

**Review by miri-bar — APPROVED** [review]

File: ai21/clients/studio/resources/chat/chat_completion.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        messages: List[ChatMessage],
+        n: Optional[int] | NotGiven = NOT_GIVEN,
+        logprobs: Optional[bool] | NotGiven = NOT_GIVEN,
+        top_logprobs: Optional[int] | NotGiven = NOT_GIVEN,
+        max_tokens: Optional[int] | NotGiven = NOT_GIVEN,
+        temperature: Optional[float] | NotGiven = NOT_GIVEN,
+        top_p: Optional[float] | NotGiven = NOT_GIVEN,
+        stop: Optional[Union[str, List[str]]] | NotGiven = NOT_GIVEN,
+        frequency_penalty: Optional[float] | NotGiven = NOT_GIVEN,
+        presence_penalty: Optional[float] | NotGiven = NOT_GIVEN,
```
Comment (etang-ai21) [inline ai21/clients/studio/resources/chat/chat_completion.py:28]: What happens if None is passed?

File: ai21/clients/studio/resources/chat/chat_completion.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        self,
+        model: str,
+        messages: List[ChatMessage],
+        n: Optional[int] | NotGiven = NOT_GIVEN,
+        logprobs: Optional[bool] | NotGiven = NOT_GIVEN,
+        top_logprobs: Optional[int] | NotGiven = NOT_GIVEN,
+        max_tokens: Optional[int] | NotGiven = NOT_GIVEN,
+        temperature: Optional[float] | NotGiven = NOT_GIVEN,
+        top_p: Optional[float] | NotGiven = NOT_GIVEN,
+        stop: Optional[Union[str, List[str]]] | NotGiven = NOT_GIVEN,
```
Comment (etang-ai21) [inline ai21/clients/studio/resources/chat/chat_completion.py:26]: |

File: ai21/clients/studio/resources/chat/chat_completion.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            messages=messages,
+            n=n,
+            logprobs=logprobs,
+            top_logprobs=top_logprobs,
+            stop=stop,
+            temperature=temperature,
+            max_tokens=max_tokens,
+            top_p=top_p,
+            frequency_penalty=frequency_penalty,
+            presence_penalty=presence_penalty,
```
Comment (etang-ai21) [inline ai21/clients/studio/resources/chat/chat_completion.py:42]: kwargs

**Comment by Asaf-m** [comment]
@asafgardin Just a reminder that the name won't be gaia. I'm still waiting for Omri to say whether it's going to be "jamba" / "jamba-small" or something else...

File: ai21/clients/studio/resources/chat/chat_completion.py (hunk 3 above)
Comment (asafgardin) [reply ai21/clients/studio/resources/chat/chat_completion.py:28]: @Asaf-m Awesome. Thanks for the heads up

File: ai21/clients/studio/resources/studio_chat.py [hunk 6]
```diff
@@ -24,6 +26,11 @@ def create(
         count_penalty: Optional[Penalty] = None,
         **kwargs,
     ) -> ChatResponse:
+        warnings.warn(
+            "This method is deprecated. Please use the `chat.completions.create` method in the client instead.",
```
Comment (grunde) [inline ai21/clients/studio/resources/studio_chat.py:30]: Let's put a pin on that. Deprecation might not happen on the first step

File: examples/studio/chat_complete.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
+from ai21 import AI21Client
+from ai21.models import RoleType
+from ai21.models.chat import ChatMessage
+
+system = "You're a support engineer in a SaaS company"
```
Comment (grunde) [inline examples/studio/chat_complete.py:5]: Shouldn't this file be nested under chat/completions?

File: ai21/clients/studio/resources/studio_chat.py (hunk 6 above)
Comment (asafgardin) [reply ai21/clients/studio/resources/studio_chat.py:30]: You mean open a ticket so we won't forget to deprecate? or to wait with the deprecation warning until we know a more specific deprecation date?

Comment (etang-ai21) [reply ai21/clients/studio/resources/studio_chat.py:30]: The latter

File: ai21/models/chat/chat_message.py [hunk 8]
```diff
@@ -0,0 +1,12 @@
+from dataclasses import dataclass
+
+from ai21.models import RoleType
```
Comment (etang-ai21) [inline ai21/models/chat/chat_message.py:3]: Maybe role type should be under `chat` and exported in `models`? This ways users won't need to guess where to import it from

File: ai21/models/chat/__init__.py [hunk 9]
```diff
@@ -0,0 +1,5 @@
+from __future__ import annotations
+
+from .chat_message import ChatMessage as ChatMessage
```
Comment (etang-ai21) [inline ai21/models/chat/__init__.py:3]: Why do you need these aliases?

File: examples/studio/chat_complete.py (hunk 7 above)
Comment (asafgardin) [reply examples/studio/chat_complete.py:5]: I put it under examples/chat/chat_completions.py

File: ai21/models/chat/__init__.py (hunk 9 above)
Comment (asafgardin) [reply ai21/models/chat/__init__.py]: `ruff` was removed imports that weren't in the main `__init__.py` saying these imports were not used, where in-fact they were. This was a workaround I saw was used. 

I found a better solution to configure ruff to not removed these types of imports from any `__init__.py` (see `pyproject.yaml`

File: pyproject.toml [hunk 10]
```diff
@@ -116,3 +116,5 @@ prerelease = true
 
 [tool.ruff]
 line-length = 120
+[tool.ruff.per-file-ignores]
+"__init__.py" = ["F401"]
```
Comment (etang-ai21) [inline pyproject.toml:120]: what's that?

Comment (asafgardin) [reply pyproject.toml:120]: Look at this comment - https://github.com/AI21Labs/ai21-python/pull/84#discussion_r1538209780

**Review by etang-ai21 — APPROVED** [review]

**Pull request merged**